[testing] - ElectronDiffraction.find_peaks() testing throws errors on correct code

**Describe the bug**
Attempts to run the find_peaks tests will sometimes throw an error even though the tested code runs fine in a notebook.


**To Reproduce**
Steps to reproduce the behavior:
1. Unskip the find_peaks_xc test in test_electron_diffraction()
2. Run test

**Stack Trace**
```
    @pytest.mark.filterwarnings('ignore::DeprecationWarning') #skimage internals
    def test_find_peaks_xc(self,diffraction_pattern):
        disc = np.ones((2,2))
>       output = diffraction_pattern.find_peaks(method='xc',disc_image=disc,min_distance=3)

tests/test_signals/test_electron_diffraction.py:208: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pyxem/signals/electron_diffraction.py:613: in find_peaks
    calibration=calib)
../../../miniconda3/envs/diffpy/lib/python3.6/site-packages/hyperspy/signal.py:3777: in map
    **kwargs)
../../../miniconda3/envs/diffpy/lib/python3.6/site-packages/hyperspy/signal.py:3927: in _map_iterate
    sig_shape)
../../../miniconda3/envs/diffpy/lib/python3.6/site-packages/hyperspy/misc/utils.py:1005: in map_result_construction
    sig.axes_manager._append_axis(sig_shape[-ind], navigate=False)
../../../miniconda3/envs/diffpy/lib/python3.6/site-packages/hyperspy/axes.py:973: in _append_axis
    axis = DataAxis(*args, **kwargs)
../../../miniconda3/envs/diffpy/lib/python3.6/site-packages/hyperspy/axes.py:264: in __init__
    self.index = 0
../../../miniconda3/envs/diffpy/lib/python3.6/site-packages/traits/trait_types.py:1865: in _set
    self.error( object, name, value )
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <traits.trait_types.Range object at 0x7faf97e201d0>, object = <[AttributeError("'DataAxis' object has no attribute 'axes_manager'") raised in repr()] DataAxis object at 0x7faf7eb6afc0>, name = 'index', value = 0

    def error ( self, object, name, value ):
        """Raises a TraitError exception.
    
            Parameters
            ----------
            object : object
                The object whose attribute is being assigned.
            name : str
                The name of the attribute being assigned.
            value : object
                The proposed new value for the attribute.
    
            Description
            -----------
            This method is called by the validate() method when an assigned value
            is not valid. Raising a TraitError exception either notifies the user of
            the problem, or, in the case of compound traits, provides a chance for
            another trait handler to handle to validate the value.
            """
        raise TraitError( object, name, self.full_info( object, name, value ),
>                         value )
E       traits.trait_errors.TraitError: The 'index' trait of a DataAxis instance must be 0 <= a number <= -1, but a value of 0 <class 'int'> was specified.

../../../miniconda3/envs/diffpy/lib/python3.6/site-packages/traits/trait_handlers.py:172: TraitError

```

--- a/pyxem/signals/electron_diffraction.py
+++ b/pyxem/signals/electron_diffraction.py
@@ -625,7 +625,7 @@ class ElectronDiffraction(Signal2D):
         x.units = 'nm'
 
         y.name = 'y'
-        y.scale = self.axes_manager.navigation_axes[0].scale
+        y.scale = self.axes_manager.navigation_axes[1].scale
         y.units = 'nm'
 
         return peaks

--- a/pyxem/utils/peakfinders2D.py
+++ b/pyxem/utils/peakfinders2D.py
@@ -341,4 +341,4 @@ def find_peaks_xc(z,disc_image,min_distance=5,peak_threshold=0.2):
     peaks = peak_local_max(response_image,min_distance=min_distance,threshold_rel=peak_threshold)
     peaks -= 1 #this means the return format is the same as the other peak finders
 
-    return peaks
+    return clean_peaks(peaks)

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -5,6 +5,9 @@ from pyxem.signals.electron_diffraction import ElectronDiffraction
 
 @pytest.fixture(scope="session",autouse=True)
 def default_structure():
+    """
+    An atomic structure represetned using diffpy
+    """
     latt = diffpy.structure.lattice.Lattice(3,3,5,90,90,120)
     atom = diffpy.structure.atom.Atom(atype='Ni',xyz=[0,0,0],lattice=latt)
     hexagonal_structure = diffpy.structure.Structure(atoms=[atom],lattice=latt)
@@ -46,4 +49,7 @@ def default_structure():
 ])
 
 def diffraction_pattern(request):
+    """
+    A boring, multiuse dp, with signature: ElectronDiffraction <2,2|8,8>
+    """
     return ElectronDiffraction(request.param)

--- a/tests/test_signals/test_electron_diffraction.py
+++ b/tests/test_signals/test_electron_diffraction.py
@@ -228,6 +228,7 @@ class TestsAssertionless:
         storage = diffraction_pattern.decomposition()
 
     @pytest.mark.filterwarnings('ignore::DeprecationWarning')
+    @pytest.mark.filterwarnings('ignore::UserWarning') #we don't want to use xc in this bit
     def test_find_peaks_interactive(self,diffraction_pattern):
         from matplotlib import pyplot as plt
         plt.ion() #to make plotting non-blocking

--- a/tests/test_signals/test_electron_diffraction.py
+++ b/tests/test_signals/test_electron_diffraction.py
@@ -138,6 +138,10 @@ class TestRadialProfile:
 
     @pytest.fixture
     def diffraction_pattern_for_radial(self):
+        """
+        Two diffraction patterns with easy to see radial profiles, wrapped
+        in ElectronDiffraction  <2|8,8>
+        """
         dp = ElectronDiffraction(np.zeros((2, 8, 8)))
         dp.data[0] = np.array([[0., 0., 2., 2., 2., 2., 0., 0.],
                                [0., 2., 3., 3., 3., 3., 2., 0.],
@@ -196,6 +200,10 @@ class TestPeakFinding:
 
     @pytest.fixture
     def ragged_peak(self):
+        """
+        A small selection of peaks in an ElectronDiffraction, to allow flexibilty
+        of test building here.
+        """
         pattern = np.zeros((2,2,128,128))
         pattern[:,:,40:42,45] = 1
         pattern[:,:,110,30:32] = 1
@@ -204,18 +212,16 @@ class TestPeakFinding:
         dp.set_diffraction_calibration(1)
         return dp
 
-    methods = ['zaefferer','laplacian_of_gaussians', 'difference_of_gaussians','stat']
+    methods = ['zaefferer','laplacian_of_gaussians', 'difference_of_gaussians','stat','xc']
 
     @pytest.mark.parametrize('method', methods)
     @pytest.mark.filterwarnings('ignore::DeprecationWarning') #skimage internals
     def test_findpeaks_ragged(self,ragged_peak,method):
-        output = ragged_peak.find_peaks(method=method,show_progressbar=False)
-
-    @pytest.mark.skip(reason="This raises a traiterror at present")
-    @pytest.mark.filterwarnings('ignore::DeprecationWarning') #skimage internals
-    def test_find_peaks_xc(self,ragged_peak):
-        disc = np.ones((2,2))
-        output = ragged_peak.find_peaks(method='xc',disc_image=disc,min_distance=3)
+        if method == 'xc':
+            disc = np.ones((2,2))
+            output = ragged_peak.find_peaks(method='xc',disc_image=disc,min_distance=3)
+        else:
+            output = ragged_peak.find_peaks(method=method,show_progressbar=False)
 
 class TestsAssertionless:
     def test_decomposition(self,diffraction_pattern):

--- a/tests/test_utils/test_expt_utils.py
+++ b/tests/test_utils/test_expt_utils.py
@@ -35,10 +35,13 @@ from scipy.ndimage.filters import gaussian_filter
               [0., 0., 0., 0., 0., 1., 0., 0.],
               [0., 0., 0., 0., 0., 0., 0., 0.]])
 ])
-def diffraction_pattern(request):
+def diffraction_pattern_one_dimension(request):
+    """
+    1D (in navigation space) diffraction pattern <1|8,8>
+    """
     return ElectronDiffraction(request.param)
 
-def test_index_coords(diffraction_pattern):
+def test_index_coords(diffraction_pattern_one_dimension):
     x = np.array([[-4., -3., -2., -1.,  0.,  1.,  2.,  3.],
                   [-4., -3., -2., -1.,  0.,  1.,  2.,  3.],
                   [-4., -3., -2., -1.,  0.,  1.,  2.,  3.],
@@ -55,12 +58,12 @@ def test_index_coords(diffraction_pattern):
                   [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                   [ 2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.],
                   [ 3.,  3.,  3.,  3.,  3.,  3.,  3.,  3.]])
-    xc, yc = _index_coords(diffraction_pattern.data)
+    xc, yc = _index_coords(diffraction_pattern_one_dimension.data)
     np.testing.assert_almost_equal(xc, x)
     np.testing.assert_almost_equal(yc, y)
 
-def test_index_coords_non_centeral(diffraction_pattern):
-    xc, yc = _index_coords(diffraction_pattern.data,origin=(0,0))
+def test_index_coords_non_centeral(diffraction_pattern_one_dimension):
+    xc, yc = _index_coords(diffraction_pattern_one_dimension.data,origin=(0,0))
     assert xc[0,0] == 0
     assert yc[0,0] == 0
     assert xc[0,5] == 5
@@ -101,8 +104,8 @@ def test_peaks_as_gvectors(z, center, calibration, g):
 
 methods = ['average','nan']
 @pytest.mark.parametrize('method', methods)
-def test_remove_dead_pixels(diffraction_pattern,method):
-    z = diffraction_pattern.data
+def test_remove_dead_pixels(diffraction_pattern_one_dimension,method):
+    z = diffraction_pattern_one_dimension.data
     dead_removed = remove_dead(z,[[3,3]],deadvalue=method)
     assert z[3,3] != dead_removed[3,3]
 

